### ISSUE TITLE
Change gameType from string to STE when connecting to master

### ISSUE THREAD

**Issue by raptor** [bitfighter/bitfighter#177]
_From [watusim...@bitfighter.org](https://code.google.com/u/105427273526970468779/) on November 26, 2012 03:06:31_

Master writeConnectRequest() should send gameType as a stringTableEntry rather than a straight string. Easy change, but will break compatibility with the master unless we're clever.

Would be open to suggestions that the cost of leaving as-is is so low and the possibility of introducing new bugs is high enough that we should just ignore.

_Original issue: http://code.google.com/p/bitfighter/issues/detail?id=177_

**Comment by raptor** [comment]
_From [buckyballreaction](https://code.google.com/u/buckyballreaction/) on June 16, 2013 21:35:01_

I'm going to not do this - I actually implemented the change (easy, not prone to bugs), then I realized a few things:

 - StringTableEntry is used to send strings that are sent more than once - the first time is always the full string
 - writeConnectRequest() is the first time the gametype string will be sent from a server, and is only ever sent once since the server only needs to connect once upon start-up
 - s2mUpdateServerStatus() is used on every level change and already uses a StringTableEntry for the gametype

Changing writeConnectRequest to use a STE for the gametype will possibly save 1 string transmission over the course of the life of the server.  I think this is not worth it.

Setting to WontFix.  Reopen if you think otherwise

**Status:** WontFix  
  **Owner:** buckyballreaction
